app/uk/gov/hmrc/helptosavefrontend/controllers/HelpToSaveAuth.scala

```diff
@@ -90,6 +103,38 @@ class HelpToSaveAuth(app: Application, frontendAuthConnector: FrontendAuthConnec
     }
   }
 
+  private def getUserInfo(nino:        String,
+                          name:        Name,
+                          email:       Option[String],
+                          dob:         Option[LocalDate],
+                          itmpName:    ItmpName,
+                          itmpDob:     Option[LocalDate],
+                          itmpAddress: ItmpAddress): Either[MissingUserInfos, UserInfo] = {
+
+    val givenNameValidation: ValidatedNel[MissingUserInfo, String] =
+      itmpName.givenName.orElse(name.name)
+        .toValidNel(MissingUserInfo.GivenName)
+
+    val surnameValidation =
+      itmpName.familyName.orElse(name.lastName)
+        .toValidNel(MissingUserInfo.Surname)
+
+    val dateOfBirthValidation =
+      itmpDob.orElse(dob)
+        .toValidNel(MissingUserInfo.DateOfBirth)
+
+    val emailValidation = email.toValidNel(MissingUserInfo.Email)
+
+    val validation: ValidatedNel[MissingUserInfo, UserInfo] =
+      (givenNameValidation |@| surnameValidation |@| dateOfBirthValidation |@| emailValidation)
+        .map((givenName, surname, jodaDob, email) ⇒
+          UserInfo(givenName, surname, nino, jodaDob, email, Address(itmpAddress)))
+
+    validation
+      .leftMap(m ⇒ MissingUserInfos(m.toList.toSet, nino))
```

Comment:
don't need to do `toList` and then `toSet` do we? Can't we go straight to `toSet`?